### DISCUSSION TITLE
How to combine both user_context and app_only authentication

### DISCUSSION THREAD

**Discussion by vxyzhang** [tweepy/tweepy#1341] [Questions]
I'm trying to get followers/friends information. I can use either user-context or app-only authentication. Either will return 15 calls in a 15 min window. But I'm wondering how to combine the two so that I can get 30 calls every 15 minutes? Thanks.

**Comment by MichaelCurrin** [comment]
I don't think it matters whether you use user-context or app-only auth, the rate limiting is still against your account.

use a combination of [paging](https://michaelcurrin.github.io/python-twitter-guide/#/code_snippets?id=paging) and auto [waiting](https://michaelcurrin.github.io/python-twitter-guide/#/auth?id=add-waiting) to avoid rate limiting.
